src/platforms/stlink/platform.c

```diff
@@ -51,6 +51,19 @@ void platform_init(void)
 #ifdef BLUEPILL
 	led_idle_run = GPIO13;
 	nrst_pin = NRST_PIN_V1;
+#elif defined(STLINK_V2_ISOL)
+	led_idle_run = GPIO9;
+	nrst_pin = NRST_PIN_V2;
+	/* PB12 is SWDIO_IN */
+	gpio_set_mode(GPIOB, GPIO_MODE_INPUT, GPIO_CNF_INPUT_FLOAT, GPIO12);
+	/* PA4 is used to set SWDCLK floating when set to 1 */
+	gpio_set_mode(GPIOA, GPIO_MODE_OUTPUT_2_MHZ, GPIO_CNF_OUTPUT_PUSHPULL, GPIO4);
+	gpio_clear(GPIOA, GPIO4);
+	/* PA1 is used to set SWDIO floating and MUXED to SWDIO_IN when set to 1 */
+	gpio_set_mode(GPIOA, GPIO_MODE_OUTPUT_2_MHZ, GPIO_CNF_OUTPUT_PUSHPULL, GPIO1);
```

Comment:
If you wanted to, you could implement feature parity with BMP v2.3 HW rev 6 w.r.t. floating the SWCLK pin, required to support stm32g0 in small pin-count packages. See Issue 945 and PR1192 (intentional not-a-mention), specifically commit https://github.com/blackmagic-debug/blackmagic/pull/1192/commits/a573c268d5cae2622c8bf659e608313c61d9801d for `native` platform.
PB12 is `T_SWDIO_IN` on **most/all** stlinkv2 schematics, and you made it a macro -- please substitute it into here (`SWDIO_IN_PIN`).
Accordingly, PA4 sounds like `TCK_DIR_PIN` and PA1 like `TMS_DIR_PIN`/`SWDIO_DIR_PIN`. This makes for less unused/magic pin numbers, even if you described their purpose in adjacent comments (nice!).